plugins/mission/mission_impl.cpp

```diff
@@ -932,4 +934,115 @@ void MissionImpl::process_timeout()
+        float speed_m_s = 0.f;
+        float gimbal_pitch_deg = 0.f, gimbal_yaw_deg = 0.f;
+        bool is_fly_through = false;
+        double lat_deg = NAN, lon_deg = NAN, rel_alt_deg = NAN;
+        double photo_interval = 0.;
+        auto camera_action = MissionItem::CameraAction::NONE;
+        auto command = json_mission_item["command"].int_value();
+
+        // Extract parameters of each mission item
+        std::vector<double> params;
+        for (auto &p : json_mission_item["params"].array_items()) {
+            params.push_back(p.number_value());
+        }
+
+        switch (command) {
+            case MAV_CMD_IMAGE_START_CAPTURE:
+                photo_interval = params[1];
+                camera_action = MissionItem::CameraAction::START_PHOTO_INTERVAL;
+                break;
+            case MAV_CMD_IMAGE_STOP_CAPTURE:
+                camera_action = MissionItem::CameraAction::STOP_PHOTO_INTERVAL;
+                break;
+            case MAV_CMD_VIDEO_START_STREAMING:
+                camera_action = MissionItem::CameraAction::START_VIDEO;
+                break;
+            case MAV_CMD_VIDEO_STOP_STREAMING:
+                camera_action = MissionItem::CameraAction::STOP_VIDEO;
+                break;
+            case MAV_CMD_DO_CHANGE_SPEED:
+                speed_m_s = params[1];
+                break;
+            case MAV_CMD_DO_MOUNT_CONTROL:
+                gimbal_pitch_deg = params[0]; // Pitch value is -ve as its in NED frame.
+                gimbal_yaw_deg = params[2];
+                break;
+            case MAV_CMD_NAV_WAYPOINT:
+                is_fly_through = (params[0] == 0.0) ? true : false;
+            // NO break here, because params 4, 5 & 6 of both
+            // of these commands are Lat, Lon & Alt respectively.
+            case MAV_CMD_NAV_TAKEOFF:
+                lat_deg = params[4];
+                lon_deg = params[5];
+                rel_alt_deg = params[6];
+                break;
+            case MAV_CMD_NAV_RETURN_TO_LAUNCH:
+                break;
+            default: // Unsupported mission command
+                LogWarn() << "Mission: Unsupported mission command: " << command << ". Ignoring.";
+                break;
+        }
+// Log disabled now; enable when necessary.
+#if 0
+        LogDebug() << "Mission: Cmd: " << command <<
+                   ", Lat: " << lat_deg << " deg, Lon: " << lon_deg << " deg, Rel alt: " << rel_alt_deg <<
+                   " meters, Speed: " << speed_m_s << " m/s, Is fly thru: " << (is_fly_through ? "true" : "false") <<
+                   " Gimbal pitch: " << gimbal_pitch_deg << " deg,  Gimbal yaw: " << gimbal_yaw_deg <<
+                   " deg, Photo interval: " << photo_interval << ", Camera action: " << MissionItem::to_str(
+                       camera_action) << "\n";
+#endif
+
+        // Add mission item to the list
+        mission_items.push_back([&]() -> std::shared_ptr<MissionItem> const {
```

Comment:
What bugs me a little bit is that the importer does not combine the mavlink mission items together into dronecore mission items.

So, for the optimal case, a location + speed command + gimbal command would optimally end up in the same dronecore mission item. Basically, all commands following a location should be cominbed with the location.